src/dom/valid-motion-props.ts

```diff
@@ -0,0 +1,61 @@
+import { MotionProps } from "../motion/types"
+
+/**
+ * A list of all valid MotionProps
+ *
+ * @internalremarks
+ * This doesn't throw if a `MotionProp` name is missing - it should.
+ */
+const validDomMotionProps = new Set<keyof MotionProps>([
+    "initial",
+    "animate",
+    "exit",
+    "style",
+    "variants",
+    "transition",
+    "transformTemplate",
+    "transformValues",
+    "custom",
+    "inherit",
+    "static",
+    "positionTransition",
+    "onAnimationComplete",
+    "onUpdate",
+    "onDragStart",
+    "onDrag",
+    "onDragEnd",
+    "onDirectionLock",
+    "onDragTransitionEnd",
+    "drag",
+    "dragConstraints",
+    "dragDirectionLock",
+    "dragElastic",
+    "dragMomentum",
+    "dragPropagation",
+    "dragTransition",
+    "dragOriginX",
+    "dragOriginY",
+    "onPan",
+    "onPanStart",
+    "onPanEnd",
+    "onPanSessionStart",
+    "onTap",
+    "onTapStart",
+    "onTapCancel",
+    "whileHover",
+    "whileTap",
+    "onHoverEnd",
+    "onHoverStart",
+])
+
+/**
+ * Check whether a prop name is a valid `MotionProp` key.
+ *
+ * @param key - Name of the property to check
+ * @returns `true` is key is a valid `MotionProp`.
+ *
+ * @public
```

Comment:
Should this be part of the public api? I'm not sure. When would a consumer of the API use this?

Reply:
Same thing we're doing, or `@emotion/is-prop-valid`